src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -627,8 +634,54 @@ public override BoundNode VisitBaseReference(BoundBaseReference node)
                 : FramePointer(node.Syntax, _topLevelMethod.ContainingType); // technically, not the correct static type
         }
 
+        private void RemapLocalFunction(CSharpSyntaxNode syntax, MethodSymbol symbol, out BoundExpression receiver, out MethodSymbol method)
+        {
+            Debug.Assert(symbol.MethodKind == MethodKind.LocalFunction);
+
+            var constructed = symbol as SubstitutedMethodSymbol;
+            if (constructed != null)
+            {
+                RemapLocalFunction(syntax, constructed.ConstructedFrom, out receiver, out method);
```

Comment:
Yes, it's partially wrong, I think. I'm not sure if this is an impossible code path or not, because local functions cannot be generic at the moment. So `symbol` can't ever be a `SubstitutedMethodSymbol`. I think this is code left over from an initial attempt at generic local functions, but I'm not sure - I'll check tomorrow when I'm back at my computer (and delete it if it's unused). In any case, this area will get rewritten once generic local functions are fully allowed.